### ISSUE TITLE
Cochran-Mantel-Haenszel

### ISSUE THREAD

**Issue by opioiddatalab** [cmrivers/epipy#19]
Thanks for this package, Caitlin. Do you have a Cochran-Mantel-Haenszel function in there? Seems like it could be useful next to the 2x2 tables. Thanks in advance!

**Comment by elofgren** [comment]
I'll see about adding this. It'll take a little thinking, because the function needs to be able to take an arbitrary number of strata to be fully functional, but it should be doable.

@cmrivers Feel free to assign this to me if you feel like it.

**Comment by elofgren** [comment]
Thinking more about this, as I'm procrastinating from doing actual work.

What's the preferred way to take in an arbitrary number of 2x2 tables, each of which is a mini-pandas data frame?

Ignoring my instinct to just make everything a numpy array, it feels like the easiest way, given there's no ordering to the strata, would be a dictionary of 2x2 tables?

**Comment by cmrivers** [comment]
Agree. You could do a pd dataframe with fancy indexing but that would be
hideous.

**Comment by elofgren** [comment]
Dictionaries it is. I'll start tinkering.

**Comment by opioiddatalab** [comment]
Thanks to both of you!

One feature that would be great to have would be to control how zero cells are handled to avoid division by zero. Biostats folks I work with sometimes add 0.1 or 0.25 to ALL 4 cells (within a stratum) to avoid division by zero. I guess in a dictionary that’s easier to control? (Although my data are in a pandas df, I think a quick collapse could get the inputs for a dictionary command.)

I think there is a Breslow-Day solution for handling ties that is widely accepted?

Thanks again!

________________________________
